src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -41,14 +50,14 @@ const NavHeader = (props) => {
     onCancelToken,
     placeholderText,
     setIsFirstTimeOpen,
-    getSearchLayout
+    getSearchLayout,
+    opacityImage
   } = props;
   const navigation = useNavigation();
   const coverPath = topicDetail?.cover_path || null;
-  const [domainHeight, setDomainHeight] = React.useState(0);
   const [searchHeight, setSearchHeight] = React.useState(0);
-  const {top} = useSafeAreaInsets();
-  const topPosition = Platform.OS === 'ios' ? top : 0;
+  const insets = useSafeAreaInsets();
+  const {width: displayWidth} = useWindowDimensions();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useSafeAreaInsets` hook is used to get the safe area insets of the device. This is a good practice as it ensures that the UI doesn't overlap with the system UI on devices with notches or rounded corners. However, I noticed that you removed the `topPosition` calculation which was previously used in the old hunk. If this calculation is necessary for your layout, consider reintroducing it.



<!-- This is an auto-generated comment by OSS CodeRabbit -->